### PR TITLE
nit(transfer): further simplify telemetry calls.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

removing additional noise from handler methods, also renames to drop reference to telemetry since the package name makes it implicit

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
Nice cleanup, @DimitrisJim!

File: modules/apps/transfer/internal/telemetry/telemetry.go [hunk 1]
```diff
@@ -11,7 +11,12 @@ import (
 	coretypes "github.com/cosmos/ibc-go/v8/modules/core/types"
 )
 
-func ReportTransferTelemetry(tokens types.Tokens, labels []metrics.Label) {
+func ReportTransfer(sourcePort, sourceChannel, destinationPort, destinationChannel string, tokens types.Tokens) {
```
Comment (crodriguezvega) [inline modules/apps/transfer/internal/telemetry/telemetry.go:14]: Should we add a doc string for these functions?

File: modules/apps/transfer/internal/telemetry/telemetry.go [hunk 2]
```diff
@@ -39,6 +58,11 @@ func ReportOnRecvPacketTelemetry(transferAmount sdkmath.Int, denomPath string, l
 		)
 	}
 
+	if token.Denom.HasPrefix(sourcePort, sourceChannel) {
+		labels = append(labels, telemetry.NewLabel(coretypes.LabelSource, "true"))
+	} else {
+		labels = append(labels, telemetry.NewLabel(coretypes.LabelSource, "false"))
+	}
```
Comment (chatton) [inline modules/apps/transfer/internal/telemetry/telemetry.go:65]:
```suggestion
		labels = append(labels, telemetry.NewLabel(coretypes.LabelSource, fmt.Sprintf("%t", token.Denom.HasPrefix(sourcePort, sourceChannel))))
```

WDYT?

**Review by chatton — APPROVED** [review]
LGTM! Big fan of incorporating package name into function calls :1st_place_medal:

File: modules/apps/transfer/keeper/relay.go [hunk 3]
```diff
@@ -271,9 +251,7 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 				return errorsmod.Wrapf(err, "failed to send coins to receiver %s", receiver.String())
 			}
 
-			denomPath := token.Denom.Path()
-			labels = append(labels, telemetry.NewLabel(coretypes.LabelSource, "false"))
-			defer internaltelemetry.ReportOnRecvPacketTelemetry(transferAmount, denomPath, labels)
+			defer telemetry.ReportOnRecvPacket(packet.GetSourcePort(), packet.GetSourceChannel(), token)
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:254]: can deduplicate even further. 

```go
If {

} else {

}
defer telemetry. ReportOnRecvPacket(...)
```

currently we execute the same defer in the if and else

**Pull request merged**